controllers/etcdpeer_controller.go

```diff
@@ -103,6 +103,76 @@ func defineReplicaSet(peer etcdv1alpha1.EtcdPeer, log logr.Logger) appsv1.Replic
 		peerLabel:    peer.Name,
 	}
 
+	etcdContainer := corev1.Container{
+		Name:  appName,
+		Image: etcdImage,
+		Env: []corev1.EnvVar{
+			{
+				Name:  etcdenvvar.InitialCluster,
+				Value: staticBootstrapInitialCluster(*peer.Spec.Bootstrap.Static),
+			},
+			{
+				Name:  etcdenvvar.Name,
+				Value: peer.Name,
+			},
+			{
+				Name:  etcdenvvar.InitialClusterToken,
+				Value: peer.Spec.ClusterName,
+			},
+			{
+				Name:  etcdenvvar.InitialAdvertisePeerURLs,
+				Value: advertiseURL(peer, etcdPeerPort).String(),
+			},
+			{
+				Name:  etcdenvvar.AdvertiseClientURLs,
+				Value: advertiseURL(peer, etcdClientPort).String(),
+			},
+			{
+				Name:  etcdenvvar.ListenPeerURLs,
+				Value: bindAllAddress(etcdPeerPort).String(),
+			},
+			{
+				Name:  etcdenvvar.ListenClientURLs,
+				Value: bindAllAddress(etcdClientPort).String(),
+			},
+			{
+				Name:  etcdenvvar.InitialClusterState,
+				Value: clusterStateValue(peer.Spec.Bootstrap.InitialClusterState),
+			},
+			{
+				Name:  etcdenvvar.DataDir,
+				Value: etcdDataMountPath,
+			},
+			// GOMAXPROCS defaults to the number of CPUs on the Kubelet host
+			// which may be much higher than the requests and limits defined for the pod,
+			// See https://github.com/golang/go/issues/33803
+			{
+				Name:  "GOMAXPROCS",
+				Value: "1",
```

Comment:
Yeah, I agree. 
Ideally, etcd would use https://github.com/uber-go/automaxprocs which would allow it to detect the limits set for the parent container.
Perhaps I can re-use that as a library to choose an appropriate value here, based on the supplied limits.

Reply:
I've updated the code to calculate GOMAXPROCS. See what you think.

Reply:
See https://github.com/etcd-io/etcd/issues/11508